pkg/informer/informer.go

```diff
@@ -532,25 +552,52 @@ func (d *DynamicDiscoverySharedInformerFactory) updateInformers() {
 			klog.V(4).InfoS("Unable to notify discovery subscriber - channel full", "id", id)
 		}
 	}
-
 }
 
 // gvrsToDiscoveryData returns "fake"/simulated discovery data for all the resources covered by the factory. It only
 // includes enough data in each APIResource to support what kcp currently needs (scheduling, placement, quota).
-func gvrsToDiscoveryData(gvrs map[schema.GroupVersionResource]struct{}) []*metav1.APIResourceList {
+func (d *DynamicDiscoverySharedInformerFactory) gvrsToDiscoveryData(gvrs map[schema.GroupVersionResource]struct{}) []*metav1.APIResourceList {
```

Comment:
A factoring that keeps this as a function, not a method, might bring clarity to the overall flow here. Gather data first, pass it in here to process. Easier to test, to boot.